enterprise/models.py

```diff
@@ -1686,6 +1691,74 @@ def sync_learner_profile_data(self):
         return identity_provider is not None and identity_provider.sync_learner_profile_data
 
 
+class EnterpriseCourseEntitlementManager(models.Manager):
+    """
+    Model manager for `EnterpriseCourseEntitlement`.
+    """
+
+    def get_queryset(self):
+        """
+        Override to return only those entitlment records for which learner is linked to an enterprise.
+        """
+        return super().get_queryset().select_related('enterprise_customer_user').filter(
+            enterprise_customer_user__linked=True
+        )
+
+
+class EnterpriseCourseEntitlement(TimeStampedModel):
+    """
+    Store the information about the entitlement of an enterprise user for a course
+    """
+
+    objects = EnterpriseCourseEntitlementManager()
+
+    class Meta:
+        unique_together = (('enterprise_customer_user', 'course_uuid',),)
+        app_label = 'enterprise'
+        ordering = ['created']
+
+    uuid = models.UUIDField(
+        primary_key=True,
+        editable=False,
+        null=False,
+        default=uuid4,
+    )
+    course_uuid = models.CharField(
+        max_length=255,
+        blank=False,
+        help_text=_(
+            "The UUID of the course (not course run) in which the learner is entitled."
+        )
+    )
+    enterprise_customer_user = models.ForeignKey(
+        EnterpriseCustomerUser,
+        blank=False,
+        null=False,
+        related_name='enterprise_entitlements',
+        on_delete=models.deletion.CASCADE,
+        help_text=_(
+            "The enterprise learner to which this entitlement is attached."
+        )
+    )
+    history = HistoricalRecords()
+
+    @cached_property
+    def course_enrollment(self):
+        """
+        Returns the ``student.CourseEnrollment`` associated with this enterprise course enrollment record.
+        """
+        if not CourseEntitlement:
+            return None
+        try:
+            return CourseEntitlement.objects.get(
+                user=self.enterprise_customer_user.user,
+                course_uuid=self.course_uuid,
+            )
+        except CourseEnrollment.DoesNotExist:
+            LOGGER.error('{} does not have a matching student.CourseEnrollment'.format(self))
+            return None
```

Comment:
nit: too much copypasta :) Should say `CourseEntitlement` everywhere.

Reply:
🤦